src/main/java/tree/ImpTargetTreeNode.kt

```diff
@@ -1,30 +1,89 @@
 package tree
 
-class ImpTargetTreeNode : TargetTreeNode {
-    override fun moveRight(): TargetTreeNode {
-        println("moveRight")
-        return this
+class ImpTargetTreeNode(
+    private val name: String = "",
+    private val tag: String = "",
+    private var id: Int = -1
+) : TargetTreeNode {
+
+    private var children: List<TargetTreeNode> = emptyList()
+    private var parent: TargetTreeNode? = null
+
+    override fun moveToRightSibling(): TargetTreeNode? {
+        val rightSibling = parent as? ImpTargetTreeNode
+        return rightSibling?.getRightSibling(this)
+    }
+
+    override fun moveToLeftSibling(): TargetTreeNode? {
+        if (parent != null)
+            return (parent!! as ImpTargetTreeNode).getLeftSibling(this)
+        return null
+    }
```

Comment:
Remove unnecessary lines
```suggestion
    override fun moveToLeftSibling(): TargetTreeNode? {
        return (parent as? ImpTargetTreeNode)?.getLeftSibling(this)
    }
```

Reply:
On it.